### PR TITLE
Add `collate` kwarg to `DataLoader`

### PR DESCRIPTION
**Pull request by lorenzoh**

### CODE REVIEW

**Comment by CarloLucibello** [comment]
Can you add more specific tests and add some for the DataLoader itself? E.g. (didn't test)
```julia
X = rand(10, 20)
d = DataLoader(X, collate=false, batchsize = 2)
for (i, x) in enumerate(d)
    @test x == [X[2i-1], X[2i]]
end

d = DataLoader(X, collate=true, batchsize = 2)
for (i, x) in enumerate(d)
    @test x == X[2i-1:2i, :]
end

d = DataLoader(X, collate=nothing, batchsize = 2)
for (i, x) in enumerate(d)
    @test x == X[2i-1:2i, :]
end
```
We should define  a custom type for testing the `nothing` vs `true` behavior.

 



```

**Comment by lorenzoh** [comment]
Added the tests 👍

File: src/eachobs.jl [hunk 1]
```diff
@@ -73,6 +73,8 @@ containing `batchsize` observations. Default `1`.
     wrapping the data container with `shuffleobs(data)`, `shuffle=true` ensures
     that the observations are shuffled anew every time you start iterating over
     `eachobs`. Default `false`.
+- `collate`: Batching behavior. See [`BatchView`](@ref) for more information. Default
+    `nothing`
```
Comment (CarloLucibello) [inline src/eachobs.jl:77]: Can you add the full docstring here? This will end up in Flux's docs

Comment (CarloLucibello) [inline src/eachobs.jl:77]:
```suggestion
- `collate`: Batching behavior. If `nothing` (default), a batch is `getobs(data, indices)`. If `false`, each batch is
    `[getobs(data, i) for i in indices]`. When `true`, applies [`batch`](@ref) to the vector of observations in a batch, 
   recursively collating arrays in the last dimensions. See [`batch`](@ref) for more information and examples.
```

**Review by CarloLucibello — APPROVED** [review]
